tests/framework/client/node.go

```diff
@@ -0,0 +1,134 @@
+		util.PanicOnError(err)
+		currentNode.Spec = node.spec
+		currentNode.Labels = node.labels
+		_, err = c.node.Update(context.Background(), currentNode, metav1.UpdateOptions{})
+		util.PanicOnError(err)
+	}
+}
+
+func (c *nodes) Get(ctx context.Context, name string, options metav1.GetOptions) (result *v1.Node, err error) {
+	return c.node.Get(ctx, name, options)
+}
+
+func (c *nodes) List(ctx context.Context, opts metav1.ListOptions) (result *v1.NodeList, err error) {
+	return c.node.List(ctx, opts)
+}
+
+func (c *nodes) Watch(ctx context.Context, opts metav1.ListOptions) (watch.Interface, error) {
+	return c.node.Watch(ctx, opts)
+}
+
+func (c *nodes) Create(ctx context.Context, node *v1.Node, opts metav1.CreateOptions) (result *v1.Node, err error) {
+	created, err := c.node.Create(ctx, node, opts)
+	if err == nil && opts.DryRun == nil {
+		createdNodes[node.Name] = nodeInfo{created.Name, ctx}
+	}
+	return created, err
+}
+
+func (c *nodes) Update(ctx context.Context, node *v1.Node, opts metav1.UpdateOptions) (result *v1.Node, err error) {
+	if opts.DryRun == nil {
+		c.saveNodeSpecs(ctx, node.Name)
+	}
+	return c.node.Update(ctx, node, opts)
+}
+
+func (c *nodes) UpdateStatus(ctx context.Context, node *v1.Node, opts metav1.UpdateOptions) (result *v1.Node, err error) {
+	return c.node.UpdateStatus(ctx, node, opts)
+}
+
+func (c *nodes) Delete(ctx context.Context, name string, opts metav1.DeleteOptions) error {
+	return c.node.Delete(ctx, name, opts)
+}
+
+func (c *nodes) DeleteCollection(ctx context.Context, opts metav1.DeleteOptions, listOpts metav1.ListOptions) error {
+	return c.node.DeleteCollection(ctx, opts, listOpts)
+}
+
+func (c *nodes) Patch(ctx context.Context, name string, pt types.PatchType, data []byte, opts metav1.PatchOptions, subresources ...string) (result *v1.Node, err error) {
+	if opts.DryRun == nil {
+		c.saveNodeSpecs(ctx, name)
+	}
+	return c.node.Patch(ctx, name, pt, data, opts, subresources...)
+}
+
+func (c *nodes) Apply(ctx context.Context, node *corev1.NodeApplyConfiguration, opts metav1.ApplyOptions) (result *v1.Node, err error) {
+	return c.node.Apply(ctx, node, opts)
+}
+
+func (c *nodes) ApplyStatus(ctx context.Context, node *corev1.NodeApplyConfiguration, opts metav1.ApplyOptions) (result *v1.Node, err error) {
+	return c.node.ApplyStatus(ctx, node, opts)
+}
+
+func (c *nodes) PatchStatus(ctx context.Context, nodeName string, data []byte) (*v1.Node, error) {
+	return c.node.PatchStatus(ctx, nodeName, data)
+}
+
+func (c *nodes) saveNodeSpecs(ctx context.Context, name string) {
+	//check that changes are not referred to a previous created node
+	if _, exist := createdNodes[name]; !exist {
+		//save the information before the first update
+		if _, exist = nodeSpecMap[name]; !exist {
+			currentNode, err := c.node.Get(ctx, name, metav1.GetOptions{})
+			util.PanicOnError(err)
+			nodeSpecMap[currentNode.Name] = nodeSpec{
+				labels: currentNode.Labels,
+				spec:   currentNode.Spec,
```

Comment:
Do we modify the spec of nodes currently?

Reply:
In the `Spec` we have `Unschedulable` and `Taints` fields, and in some tests we modify them. Instead of store the specific fields I opted to store all the `Spec`

Reply:
I modified the `nodeSpec` struct, using `Unschedulable` and `Taints` directly, avoiding storing other stuff we are not interested in. Thanks